File: net/management-api-v2/PostSpace.cs
// DocSection: cm_api_v2_post_space
// Tip: Find more about .NET SDKs at https://kontent.ai/learn/net
using Kontent.Ai.Management;

var client = new ManagementClient(new ManagementOptions
{
    ApiKey = "<YOUR_API_KEY>",
    ProjectId = "<YOUR_ENVIRONMENT_ID>"
});

var response = await client.CreateSpaceAsync(new SpaceCreateModel
    {
        Name = "Space 1", 
        Codename = "space_1",
        WebSpotlightRootItem = Reference.ByCodename("root_item_1")
    }
);
// EndDocSection

File: js/management-api-v2/PostSpace.js
// DocSection: cm_api_v2_post_space
// Tip: Find more about JS/TS SDKs at https://kontent.ai/learn/javascript
// Using ES6 syntax
import { ManagementClient } from '@kontent-ai/management-sdk';

const client = new ManagementClient({
  environmentId: '<YOUR_ENVIRONMENT_ID>',
  apiKey: '<YOUR_API_KEY>'
});

const response = await client
  .addSpace()
  .withData(
    {
      name: "my new_space",
      codename: "my_new_space"
    }
  )
  .toPromise();
// EndDocSection